sympy/stats/random_matrix_models.py

```diff
@@ -212,6 +214,45 @@ def joint_eigen_distribution(mat):
     """
     return mat.pspace.model.joint_eigen_distribution()
 
+@dispatch(Matrix)
+def joint_eigen_distribution(mat):
+    """
+    Creates joint distribution of eigen values of matrices with random
+    expressions.
+
+    Parameters
+    ==========
+
+    mat: Matrix
+        The matrix under consideration
+
+    Returns
+    =======
+
+    JointDistributionHandmade
+
+    Examples
+    ========
+
+    >>> from sympy.stats import Normal, joint_eigen_distribution
+    >>> from sympy import Matrix
+    >>> A = [[Normal('A00', 0, 1), Normal('A01', 0, 1)],
+    ... [Normal('A10', 0, 1), Normal('A11', 0, 1)]]
+    >>> joint_eigen_distribution(Matrix(A))
+    JointDistributionHandmade(-sqrt(A00**2 - 2*A00*A11 + 4*A01*A10 + A11**2)/2
+    + A00/2 + A11/2, sqrt(A00**2 - 2*A00*A11 + 4*A01*A10 + A11**2)/2 + A00/2 + A11/2)
+
+    """
+    eigenvals, counts = mat.eigenvals().keys(), mat.eigenvals().values()
+    if any(not eigenval.has(RandomSymbol) for eigenval in eigenvals):
+        raise ValueError("Eigen values don't have any random expression, "
+                         "joint distribution cannot be generated.")
+    expanded = []
+    for eigenval, count in zip(eigenvals, counts):
+        l = [eigenval]*count
+        expanded.extend(l)
```

Comment:
If you want to do it in a single line:
```py
expanded = [e for ev, cnt in zip(eigenvals, counts) for e in [ev]*cnt]
```

Reply:
Ah! I was using unpacking through `*`, that's why it didn't work. Thanks for this idea. I will update.